### ISSUE TITLE
Mount command failed

### ISSUE THREAD

**Issue by Yekutiel** [rear/rear#786]
### rear version (/usr/sbin/rear -V)

Relax-and-Recover 1.17.2 / Git
### OS version (cat /etc/rear/os.conf or lsb_release  -a)

Distributor ID: Ubuntu
Description: GalliumOS 1.0beta2
Release: 15.04
Codename: vivid
### rear configuration files (cat /etc/rear/site.conf or cat /etc/rear/local.conf)

OUTPUT=USB
BACKUP=NETFS
BACKUP_URL=usb:///dev/disk/by-label/REAR-000
### Brief description of the issue

**I copied and pasted the following from a Terminal:**

Relax-and-Recover 1.17.2 / Git
Using log file: /home/ken/rear/var/log/rear/rear-ken-Peppy.log
ERROR: Mount command 'mount -v -o rw,noatime /dev/disk/by-label/REAR-000 /tmp/rear.w44KJUW6vBZaFjO/outputfs' failed.
Aborting due to an error, check /home/ken/rear/var/log/rear/rear-ken-Peppy.log for details

**I copied and pasted the following text from/home/ken/rear/var/log/rear/rear-ken-Peppy.log:**

2016-02-25 10:47:08 Relax-and-Recover 1.17.2 / Git
2016-02-25 10:47:08 Command line options: usr/sbin/rear -v mkrescue
2016-02-25 10:47:08 Using log file: /home/ken/rear/var/log/rear/rear-ken-Peppy.log
2016-02-25 10:47:08 Including conf/Linux-i386.conf
2016-02-25 10:47:08 Including conf/GNU/Linux.conf
2016-02-25 10:47:08 Including conf/Ubuntu.conf
2016-02-25 10:47:08 Including /home/ken/rear/etc/rear/local.conf
2016-02-25 10:47:08 Running 'init' stage
2016-02-25 10:47:08 Including init/default/01_set_drlm_env.sh
2016-02-25 10:47:08 Including init/default/05_check_rear-recover_mode.sh
2016-02-25 10:47:08 Finished running 'init' stage in 0 seconds
2016-02-25 10:47:08 Using build area '/tmp/rear.w44KJUW6vBZaFjO'
mkdir: created directory '/tmp/rear.w44KJUW6vBZaFjO/rootfs'
mkdir: created directory '/tmp/rear.w44KJUW6vBZaFjO/tmp'
2016-02-25 10:47:08 Running mkrescue workflow
2016-02-25 10:47:08 Running 'prep' stage
2016-02-25 10:47:08 Including prep/default/00_remove_workflow_conf.sh
mkdir: created directory '/tmp/rear.w44KJUW6vBZaFjO/rootfs/etc'
mkdir: created directory '/tmp/rear.w44KJUW6vBZaFjO/rootfs/etc/rear'
2016-02-25 10:47:08 Including prep/default/02_translate_url.sh
2016-02-25 10:47:08 Including prep/default/03_translate_tape.sh
2016-02-25 10:47:08 Including prep/default/04_check_output_scheme.sh
2016-02-25 10:47:08 Including prep/USB/default/04_set_NETFS_requirements.sh
2016-02-25 10:47:08 Including prep/NETFS/default/05_check_NETFS_requirements.sh
2016-02-25 10:47:08 Skipping ping test
2016-02-25 10:47:08 Including prep/default/05_check_keep_old_output_copy_var.sh
2016-02-25 10:47:08 Including prep/NETFS/default/06_mount_NETFS_path.sh
mkdir: created directory '/tmp/rear.w44KJUW6vBZaFjO/outputfs'
2016-02-25 10:47:08 Mounting with 'mount -v -o rw,noatime /dev/disk/by-label/REAR-000 /tmp/rear.w44KJUW6vBZaFjO/outputfs'
mount: special device /dev/disk/by-label/REAR-000 does not exist
2016-02-25 10:47:08 ERROR: Mount command 'mount -v -o rw,noatime /dev/disk/by-label/REAR-000 /tmp/rear.w44KJUW6vBZaFjO/outputfs' failed.
=== Stack trace ===
Trace 0: usr/sbin/rear:410 main
Trace 1: /home/ken/rear/usr/share/rear/lib/mkrescue-workflow.sh:25 WORKFLOW_mkrescue
Trace 2: /home/ken/rear/usr/share/rear/lib/framework-functions.sh:85 SourceStage
Trace 3: /home/ken/rear/usr/share/rear/lib/framework-functions.sh:45 Source
Trace 4: /home/ken/rear/usr/share/rear/prep/NETFS/default/06_mount_NETFS_path.sh:11 source
Trace 5: /home/ken/rear/usr/share/rear/lib/global-functions.sh:163 mount_url
Trace 6: /home/ken/rear/usr/share/rear/lib/_input-output-functions.sh:150 StopIfError
# Message: Mount command `mount -v -o rw,noatime /dev/disk/by-label/REAR-000 /tmp/rear.w44KJUW6vBZaFjO/outputfs` failed.

2016-02-25 10:47:08 Running exit tasks.
rmdir: removing directory, '/tmp/rear.w44KJUW6vBZaFjO/outputfs'
2016-02-25 10:47:08 Finished in 0 seconds
2016-02-25 10:47:08 Removing build area /tmp/rear.w44KJUW6vBZaFjO
removed directory: '/tmp/rear.w44KJUW6vBZaFjO'
2016-02-25 10:47:08 End of program reached
### My use case

Based on skimming a few of the issues in this forum I suppose that the most of the folks submitting issues for “rear” are sophisticated Linux users, probably developers, who like solving thorny technical problems. I am a novice Linux user who prefers to avoid dealing with technical problems.

I have one particular need I would like your help with: I want to clone my installation of GalliumOS [A fast and lightweight Linux distro for ChromeOS devices.](https://galliumos.org/) on a Acer C720 to a second (different) Acer C720. 

In other words, I want to avoid having to spend a few hours reconfiguring a fresh installation of Gallium OS so that it is customized to my taste. That’s it. In other words, I am **not** a Linux sys admin responsible for a complex heterogeneous network. I have already perused the web for a few hours. I found pages such as the following: [Easiest way to move complete linux install from one SSD to another?](https://www.reddit.com/r/linux/comments/37uoya/easiest_way_to_move_complete_linux_install_from/). But I still haven't implemented an effective, let alone elegant, solution.

I tried “dd”, Clonezilla, and Redo Backup and Recovery. “dd” sort of worked but not quite. I used “dd” to clone the SSD from the first Acer C720 to a USB stick. I then inserted the USB stick into a second (different) Acer C720. I then used “dd” to clone the data from the USB stick to the SSD of the second Acer C720. I then restarted the second Acer C720. It worked fairly well. I was actually pleasantly surprised initially. But I experienced too much glitchy behavior that would likely become very annoying if I were to need to deal with it on a day-to-day basis. As for Clonezilla and [Redo Backup and Recovery](http://redobackup.org/), I failed to create a backup with both of them.

Frankly, I would ideally like to find an application that works like [UNetbootin](http://unetbootin.github.io/) in reverse and in particular (not like UNetbootin which, for the sake of this analogy, works sort of forward and in general)  which has a very simple graphical user interface. But I don't mind copying and pasting a set of straightforward commands into a Terminal.

**Comment by gdha** [comment]
For some reason `udevd` did not create the symbolic link to `/dev/disk/by-label/REAR-000` from your USB device. However, as a work-around you could use `BACKUP_URL=usb:///dev/sdc1` instead (if sdc1 is your USB device of course).

**Comment by Yekutiel** [comment]
Thanks for your help. 

I followed your instructions.  As a work-around I used BACKUP_URL=usb:///dev/sdb1 (because sdb1 is my USB device). Below are the error messages I received this time:

**I copied and pasted the following from a Terminal:**

ken@ken-Peppy:~/rear$ sudo usr/sbin/rear -v mkrescue
Relax-and-Recover 1.17.2 / Git
Using log file: /home/ken/rear/var/log/rear/rear-ken-Peppy.log
ERROR: Could not find 'mbr.bin' in . Syslinux version 3.08 or newer is required, 4.x prefered !
Aborting due to an error, check /home/ken/rear/var/log/rear/rear-ken-Peppy.log for details

**I copied and pasted the following text from/home/ken/rear/var/log/rear/rear-ken-Peppy.log:**

2016-03-02 17:03:20 Relax-and-Recover 1.17.2 / Git
2016-03-02 17:03:20 Command line options: usr/sbin/rear -v mkrescue
2016-03-02 17:03:20 Using log file: /home/ken/rear/var/log/rear/rear-ken-Peppy.log
2016-03-02 17:03:20 Including conf/Linux-i386.conf
2016-03-02 17:03:20 Including conf/GNU/Linux.conf
2016-03-02 17:03:20 Including conf/Ubuntu.conf
2016-03-02 17:03:20 Including /home/ken/rear/etc/rear/local.conf
2016-03-02 17:03:20 Running 'init' stage
2016-03-02 17:03:20 Including init/default/01_set_drlm_env.sh
2016-03-02 17:03:20 Including init/default/05_check_rear-recover_mode.sh
2016-03-02 17:03:20 Finished running 'init' stage in 0 seconds
2016-03-02 17:03:20 Using build area '/tmp/rear.p5TngMnQ32XcnQ2'
mkdir: created directory '/tmp/rear.p5TngMnQ32XcnQ2/rootfs'
mkdir: created directory '/tmp/rear.p5TngMnQ32XcnQ2/tmp'
2016-03-02 17:03:20 Running mkrescue workflow
2016-03-02 17:03:20 Running 'prep' stage
2016-03-02 17:03:20 Including prep/default/00_remove_workflow_conf.sh
mkdir: created directory '/tmp/rear.p5TngMnQ32XcnQ2/rootfs/etc'
mkdir: created directory '/tmp/rear.p5TngMnQ32XcnQ2/rootfs/etc/rear'
2016-03-02 17:03:20 Including prep/default/02_translate_url.sh
2016-03-02 17:03:20 Including prep/default/03_translate_tape.sh
2016-03-02 17:03:20 Including prep/default/04_check_output_scheme.sh
2016-03-02 17:03:20 Including prep/USB/default/04_set_NETFS_requirements.sh
2016-03-02 17:03:20 Including prep/NETFS/default/05_check_NETFS_requirements.sh
2016-03-02 17:03:20 Skipping ping test
2016-03-02 17:03:20 Including prep/default/05_check_keep_old_output_copy_var.sh
2016-03-02 17:03:20 Including prep/NETFS/default/06_mount_NETFS_path.sh
mkdir: created directory '/tmp/rear.p5TngMnQ32XcnQ2/outputfs'
2016-03-02 17:03:20 Mounting with 'mount -v -o rw,noatime /dev/sdb1 /tmp/rear.p5TngMnQ32XcnQ2/outputfs'
mount: /dev/sdb1 mounted on /tmp/rear.p5TngMnQ32XcnQ2/outputfs.
2016-03-02 17:03:20 Including prep/USB/default/06_set_usb_device.sh
2016-03-02 17:03:20 Including prep/NETFS/default/07_set_backup_archive.sh
2016-03-02 17:03:20 Including prep/NETFS/default/09_check_encrypted_backup.sh
2016-03-02 17:03:20 Including prep/default/10_init_workflow_conf.sh
2016-03-02 17:03:20 Including prep/NETFS/default/15_save_rsync_version.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/20_include_getty.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/20_include_serial_console.sh
2016-03-02 17:03:20 Including prep/NETFS/GNU/Linux/20_selinux_in_use.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/21_include_dhclient.sh
2016-03-02 17:03:20 Running DHCP client found, enabling USE_DHCLIENT
2016-03-02 17:03:20 Including prep/GNU/Linux/22_include_lvm_tools.sh
2016-03-02 17:03:20 Device mapper found enabled. Including LVM tools.
2016-03-02 17:03:20 Including prep/GNU/Linux/23_include_md_tools.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/24_include_multipath_tools.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/28_include_systemd.sh
2016-03-02 17:03:20 Including systemd (init replacement) tool-set to bootstrap Relax-and-Recover
2016-03-02 17:03:20 Including prep/GNU/Linux/28_include_vmware_tools.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/29_include_drbd.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/30_check_backup_and_output_url.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/30_include_grub_tools.sh
2016-03-02 17:03:20 Including prep/GNU/Linux/31_include_cap_uitls.sh
2016-03-02 17:03:20 Including prep/default/31_include_uefi_tools.sh
2016-03-02 17:03:20 Including prep/default/32_include_uefi_env.sh
2016-03-02 17:03:20 Including prep/USB/Linux-i386/34_find_mbr_bin.sh
dirname: missing operand
Try 'dirname --help' for more information.
2016-03-02 17:03:20 ERROR: Could not find 'mbr.bin' in . Syslinux version 3.08 or newer is required, 4.x prefered !
=== Stack trace ===
Trace 0: usr/sbin/rear:410 main
Trace 1: /home/ken/rear/usr/share/rear/lib/mkrescue-workflow.sh:25 WORKFLOW_mkrescue
Trace 2: /home/ken/rear/usr/share/rear/lib/framework-functions.sh:85 SourceStage
Trace 3: /home/ken/rear/usr/share/rear/lib/framework-functions.sh:45 Source
Trace 4: /home/ken/rear/usr/share/rear/prep/USB/Linux-i386/34_find_mbr_bin.sh:6 source
Trace 5: /home/ken/rear/usr/share/rear/lib/_input-output-functions.sh:150 StopIfError
Message: Could not find 'mbr.bin' in . Syslinux version 3.08 or newer is required, 4.x prefered !
===================
2016-03-02 17:03:20 Running exit tasks.
umount: /tmp/rear.p5TngMnQ32XcnQ2/outputfs (/dev/sdb1) unmounted
rmdir: removing directory, '/tmp/rear.p5TngMnQ32XcnQ2/outputfs'
2016-03-02 17:03:20 Finished in 0 seconds
2016-03-02 17:03:20 Removing build area /tmp/rear.p5TngMnQ32XcnQ2
removed directory: '/tmp/rear.p5TngMnQ32XcnQ2'
2016-03-02 17:03:20 End of program reached

**Comment by gdha** [comment]
Did you install syslinux and syslinux-extlinux which you need to make an USB device bootable?

**Comment by Yekutiel** [comment]
Thanks for following up with me.

I have not knowingly install syslinux and syslinux-extlinux.

I did use Gparted to format my USB device and set the flag to boot. Wouldn't that make my USB device bootable?

**Comment by gdha** [comment]
@Yekutiel Rear is doing that for you - I wouldn't use gparted for it.

**Comment by Yekutiel** [comment]
I see. How do I install syslinux and syslinux-extlinux a USB device bootable?

**Comment by gdha** [comment]
@Yekutiel Use `apt-get install syslinux` and `apt-get install extlinux` before running rear. Within rear format the USB disk as `rear format /dev/sdc`  (where /dev/sdc is your USB disk - change this if required)

**Comment by Yekutiel** [comment]
Thanks again for your help. I successfully ran `apt-get install syslinux` and `apt-get install extlinux`. That was trivial except for realizing I needed to prepend `sudo`.

I guess maybe it might be helpful to others if that information were up on http://relax-and-recover.org/documentation/getting-started

I was unsure what this means:

> Within rear

In a Terminal I entered `cd rear/` `rear format /dev/sdb` (my USB drive is **sdB** which I verified by opening Gparted to check my drive names). I also tried `rear format /dev/sdb` without changing to the rear directory.

As I suppose you realize, neither seemed to work. I took a shot in the dark... and apparently I missed.

I poked around a bit and noticed the following on my internal SSD **/home/ken/rear/usr/share/rear/format/USB/default/** but that seems to contain information Rear uses, not an interface for a user to run.

By the way, I was able to boot from the USB drive to what appeared to be the "standard Rear interface" I was hoping to see a menu choice that said, "If you want to back up your entire internal drive to a removable drive, choose this!"

I'm a novice who aspires to remain a novice in this domain yet back up his entire SSD to a USB. Like most novices who are uninterested in technical details, I want simple choices. If it takes 2 hours instead of 2 minutes for my SSD to back up to my USB that doesn't bother me much at all. In this case I care about ease and effectiveness, not efficiency.

I hope I do not offend you by opining that I suppose Rear might become very widely used by ordinary Linux users if it had a very easy way for ordinary Linux users to back up their entire internal storage devices to a removable storage device. I suppose doing this on Rear might seem very easy to you, but you are almost certainly not an ordinary Linux user. I can imagine Rear adding a menu to Gparted that said something like: "Disaster Recovery." Under that would be a choice: "Create a backup." Then users would choose a source drive and a destination drive.

Sure, what I am proposing would be a  very different project than Rear. But it would be much for accessible for those of us non-technical Linux users.
